integration/control_connection_test.rb

```diff
@@ -0,0 +1,93 @@
+# encoding: utf-8
+
+#--
+# Copyright 2013-2016 DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#++
+
+require File.dirname(__FILE__) + '/integration_test_case.rb'
+
+class ControlConnectionTest < IntegrationTestCase
+
+  def self.before_suite
+    @@ccm_cluster = CCM.setup_cluster(1, 2)
+  end
+
+  def self.after_suite
+    CCM.remove_cluster(@@ccm_cluster.name)
+  end
+
+  def remove_peer_info(info)
+    # Make sure to only connect to node1, as each node has its own peers table
+    allowed_ips = ['127.0.0.1']
+    round_robin = Cassandra::LoadBalancing::Policies::RoundRobin.new
+    whitelist = Cassandra::LoadBalancing::Policies::WhiteList.new(allowed_ips, round_robin)
+    cluster = Cassandra.cluster(load_balancing_policy: whitelist)
+    session = cluster.connect
+
+    value = session.execute("SELECT #{info} FROM system.peers WHERE peer = '127.0.0.2'").first["#{info}"]
+    session.execute("DELETE #{info} FROM system.peers WHERE peer = '127.0.0.2'")
+    result = session.execute("SELECT #{info} FROM system.peers WHERE peer = '127.0.0.2'").first
+    assert_nil result["#{info}"]
+
+    cluster.close
+    value
+  end
+
+  def restore_peer_info(info, value)
+    # Make sure to only connect to node1, as each node has its own peers table
+    allowed_ips = ['127.0.0.1']
+    round_robin = Cassandra::LoadBalancing::Policies::RoundRobin.new
+    whitelist = Cassandra::LoadBalancing::Policies::WhiteList.new(allowed_ips, round_robin)
+    cluster = Cassandra.cluster(load_balancing_policy: whitelist)
+    session = cluster.connect
+
+    update = session.prepare("UPDATE system.peers SET #{info}=? WHERE peer = '127.0.0.2'")
```

Comment:
Not a big deal, but preparing the statement doesn't buy you anything in this case. You could just do session.execute("UPDATE ...", arguments: [value])